### PR TITLE
[CLI-2958] Standardize `--ca-location` and `--ca-cert-path`

### PR DESCRIPTION
**Pull request by tmalikconfluent**
Release Notes
-------------

Breaking Changes
- Flags `--ca-location` and `--ca-cert-path` have been changed to `--certificate-authority-path` across the CLI.
- `CONFLUENT_PLATFORM_CA_CERT_PATH` changed to `CONFLUENT_PLATFORM_CERTIFICATE_AUTHORITY_PATH`

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----
Standardize `--ca-location` and `--ca-cert-path`

References
----------
https://confluentinc.atlassian.net/browse/CLI-2958

Test & Review
-------------
Updated the integration tests and subsequent golden files.

### CODE REVIEW

File: internal/cluster/command_describe.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func getURL(cmd *cobra.Command) (string, error) {
 
 func getCACertPath(cmd *cobra.Command) (string, error) {
 	// Order of precedence: flags > env vars
-	if caCertPath, err := cmd.Flags().GetString("ca-cert-path"); caCertPath != "" || err != nil {
+	if caCertPath, err := cmd.Flags().GetString("certificate-authority-path"); caCertPath != "" || err != nil {
```
Comment (brianstrauch) [inline internal/cluster/command_describe.go:95]:
```suggestion
	if certificateAuthorityPath, err := cmd.Flags().GetString("certificate-authority-path"); certificateAuthorityPath != "" || err != nil {
```

File: internal/kafka/confluent_kafka_configs.go [hunk 2]
```diff
@@ -107,7 +107,7 @@ func getOnPremProducerConfigMap(cmd *cobra.Command, clientID string) (*ckafka.Co
 		return nil, err
 	}
 	if protocol == "SSL" || protocol == "SASL_SSL" {
-		caLocation, err := cmd.Flags().GetString("ca-location")
+		caLocation, err := cmd.Flags().GetString("certificate-authority-path")
```
Comment (brianstrauch) [inline internal/kafka/confluent_kafka_configs.go:110]:
```suggestion
		certificateAuthorityPath, err := cmd.Flags().GetString("certificate-authority-path")
```

File: internal/kafka/confluent_kafka_configs.go [hunk 3]
```diff
@@ -147,7 +147,7 @@ func getOnPremConsumerConfigMap(cmd *cobra.Command, clientID string) (*ckafka.Co
...
```
Comment (brianstrauch) [inline internal/kafka/confluent_kafka_configs.go:150]:
```suggestion
		certificateAuthorityPath, err := cmd.Flags().GetString("certificate-authority-path")
```

File: internal/login/command.go [hunk 4]
```diff
@@ -289,15 +289,15 @@ func (c *command) loginMDS(cmd *cobra.Command, url string) error {
...
 // For legacy users:
-// if ca-cert-path flag is not used, then return caCertPath value stored in config for the login context
-// if user passes empty string for ca-cert-path flag then reset the ca-cert-path value in config for the context
-// (only for legacy contexts is it still possible for the context name without ca-cert-path to have ca-cert-path)
+// if certificate-authority-path flag is not used, then return caCertPath value stored in config for the login context
+// if user passes empty string for certificate-authority-path flag then reset the certificate-authority-path value in config for the context
+// (only for legacy contexts is it still possible for the context name without certificate-authority-path to have certificate-authority-path)
 func (c *command) getCaCertPath(cmd *cobra.Command, username, url string) (string, bool, error) {
-	caCertPath, err := cmd.Flags().GetString("ca-cert-path")
+	caCertPath, err := cmd.Flags().GetString("certificate-authority-path")
```
Comment (brianstrauch) [inline internal/login/command.go:300]:
```suggestion
	certificateAuthorityPath, err := cmd.Flags().GetString("certificate-authority-path")
```

File: internal/login/command.go [hunk 5]
```diff
@@ -372,8 +372,8 @@ func (c *command) getConfluentCredentials(cmd *cobra.Command, url string) (*paut
 }
 
 func (c *command) checkLegacyContextCACertPath(cmd *cobra.Command, contextName string) string {
-	changed := cmd.Flags().Changed("ca-cert-path")
-	// if flag used but empty string is passed then user intends to reset the ca-cert-path
+	changed := cmd.Flags().Changed("certificate-authority-path")
+	// if flag used but empty string is passed then user intends to reset the certificate-authority-path
 	if changed {
```
Comment (brianstrauch) [inline internal/login/command.go:377]:
```suggestion
	// if flag used but empty string is passed then user intends to reset the certificate-authority-path
	if cmd.Flags().Changed("certificate-authority-path") {
```

File: pkg/cmd/authenticated_cli_command.go [hunk 6]
```diff
@@ -188,7 +188,7 @@ func (c *AuthenticatedCLICommand) GetSchemaRegistryClient(cmd *cobra.Command) (*
 			}
 			configuration.Servers = srsdk.ServerConfigurations{{URL: u.String()}}
 
-			caLocation, err := cmd.Flags().GetString("ca-location")
+			caLocation, err := cmd.Flags().GetString("certificate-authority-path")
```
Comment (brianstrauch) [inline pkg/cmd/authenticated_cli_command.go:191]:
```suggestion
			certificateAuthorityPath, err := cmd.Flags().GetString("certificate-authority-path")
```

File: pkg/errors/error_message.go [hunk 7]
```diff
@@ -126,7 +126,7 @@ const (
 	InternalServerErrorSuggestions    = "Please check the status of your Kafka cluster or submit a support ticket."
 	KafkaRestErrorMsg                 = "Kafka REST request failed: %s %s: %s"
 	KafkaRestConnectionErrorMsg       = "unable to establish Kafka REST connection: %s: %s"
-	KafkaRestCertErrorSuggestions     = "To specify a CA certificate, please use the `--ca-cert-path` flag or set `CONFLUENT_PLATFORM_CA_CERT_PATH`."
+	KafkaRestCertErrorSuggestions     = "To specify a CA certificate, please use the `--certificate-authority-path` flag or set `CONFLUENT_PLATFORM_CA_CERT_PATH`."
```
Comment (brianstrauch) [inline pkg/errors/error_message.go:129]: These should probably match.
```suggestion
	KafkaRestCertErrorSuggestions     = "To specify a CA certificate, please use the `--certificate-authority-path` flag or set `CONFLUENT_PLATFORM_CERTIFICATE_AUTHORITY_PATH`."
```

File: test/fixtures/output/kafka/acl/create-help-onprem.golden [hunk 8]
```diff
@@ -23,24 +23,24 @@ You can also run the example above without logging in if you provide the Kafka R
...
+      --allow                               ACL permission to allow access.
+      --deny                                ACL permission to restrict access to resource.
+      --cluster-scope                       Set the cluster resource. With this option the ACL grants access to the provided operations on the Kafka cluster itself.
+      --consumer-group string               Set the Consumer Group resource.
+      --transactional-id string             Set the TransactionalID resource.
+      --topic string                        Set the topic resource. With this option the ACL grants the provided operations on the topics that start with that prefix, depending on whether the --prefix option was also passed.
+      --prefix                              Set to match all resource names prefixed with this value.
+      --url string                          Base URL of REST Proxy Endpoint of Kafka Cluster (include "/kafka" for embedded Rest Proxy). Must set flag or CONFLUENT_REST_URL.
+      --certificate-authority-path string   Path to a PEM-encoded CA to verify the Confluent REST Proxy.
```
Comment (brianstrauch) [inline test/fixtures/output/kafka/acl/create-help-onprem.golden:37]:
```suggestion
      --certificate-authority-path string   Path to a PEM-encoded Certificate Authority to verify the Confluent REST Proxy.
```

File: pkg/errors/error_message.go [hunk 9]
```diff
@@ -126,7 +126,7 @@ const (
 	InternalServerErrorSuggestions    = "Please check the status of your Kafka cluster or submit a support ticket."
 	KafkaRestErrorMsg                 = "Kafka REST request failed: %s %s: %s"
 	KafkaRestConnectionErrorMsg       = "unable to establish Kafka REST connection: %s: %s"
-	KafkaRestCertErrorSuggestions     = "To specify a CA certificate, please use the `--ca-cert-path` flag or set `CONFLUENT_PLATFORM_CA_CERT_PATH`."
+	KafkaRestCertErrorSuggestions     = "To specify a Certificate Authority certificate, please use the `--certificate-authority-path` flag or set `CONFLUENT_PLATFORM_CERTIFICATE_AUTHORITY_PATH`."
```
Comment (brianstrauch) [inline pkg/errors/error_message.go:129]: Make sure you update all the other instances of `CONFLUENT_PLATFORM_CERTIFICATE_AUTHORITY_PATH`!

**Review by brianstrauch — CHANGES_REQUESTED** [review]

File: pkg/errors/error_message.go (hunk 9 above)
Comment (tmalikconfluent) [reply pkg/errors/error_message.go:129]: done

**Review by brianstrauch — APPROVED** [review]

**Comment by brianstrauch** [comment]
Make sure you update the release notes to mention the environment variable change. Thanks for the PR!

File: cmd/lint/main.go [hunk 10]
```diff
@@ -112,6 +112,7 @@ var flagRules = []linter.FlagRule{
 			"source-bootstrap-server",
 			"update-schema-registry",
 			"worker-configurations",
+			"certificate-authority-path",
```
Comment (brianstrauch) [inline cmd/lint/main.go:115]: Please insert in alphabetical order so it's easier to find it as this list grows!

Comment (brianstrauch) [reply cmd/lint/main.go:115]: (and while you're at it, can you alphabetize the "remote-*" flags too?)

**Pull request merged**